src/qibo/gates/gates.py

```diff
@@ -1049,6 +1055,117 @@ def parameters(self, x):
         ParametrizedGate.parameters.fset(self, x)  # pylint: disable=no-member
 
 
+class _Rnn_(ParametrizedGate):
+    """Abstract class for defining the RXX, RYY and RZZ rotations.
+
+    Args:
+        q0 (int): the first entangled qubit id number.
+        q1 (int): the second entangled qubit id number.
+        theta (float): the rotation angle.
+        trainable (bool): whether gate parameters can be updated using
+            :meth:`qibo.models.circuit.AbstractCircuit.set_parameters`
+            (default is ``True``).
+    """
+
+    def __init__(self, q0, q1, theta, trainable=True):
+        super().__init__(trainable)
+        self.name = None
+        self._controlled_gate = None
+        self.target_qubits = (q0, q1)
+
+        self.parameters = theta
+        self.init_args = [q0, q1]
+        self.init_kwargs = {"theta": theta, "trainable": trainable}
+
+    def _dagger(self) -> "Gate":
+        """"""
+        return self.__class__(
```

Comment:
@damarkian thanks for this. Could you please include some test which covers this return statement?